src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -769,6 +769,11 @@ public int Step(IDebugThread2 pThread, enum_STEPKIND kind, enum_STEPUNIT unit)
 
             try
             {
+                if (null == thread || null == thread.GetDebuggedThread())
+                {
+                    return Constants.E_FAIL;
```

Comment:
Kneejerk reaction to E_FAIL (like we spoke about). Concord winds returning a xapi failed hr from the dispatcher in this case (soo concord\impl\ad7\Programm.cpp step, so there is precedence here.